source/nodejs/adaptivecards-designer/src/designer-peers.ts

```diff
@@ -2424,16 +2537,20 @@ export class ImagePeer extends TypedCardElementPeer<Adaptive.Image> {
 export class MediaPeer extends TypedCardElementPeer<Adaptive.Media> {
     static readonly altTextProperty = new StringPropertyEditor(Adaptive.Versions.v1_1, "altText", "Alternate text", true);
     static readonly posterUrlProperty = new StringPropertyEditor(Adaptive.Versions.v1_1, "posterUrl", "Poster URL", true);
-    static readonly sourcesProperty = new NameValuePairPropertyEditor(
+    static readonly sourcesProperty = new InnerStructPropertyEditor(
         Adaptive.Versions.v1_1,
         "sources",
-        "url",
-        "mimeType",
-        (name: string, value: string) => { return new Adaptive.MediaSource(name, value); },
-        "URL",
-        "MIME type",
+        {"url": {name: "URL"}, "mimeType": {name: "MIME Type"}},
```

Comment:
For the mineType, I'm wondering if it's better to use combobox to avoid wrong input

Reply:
We did not use a combo box for `MediaSource` before, so to maintain consistency I think it is okay for `CaptionSource` to also use a text box. If we want a combo box for this, it can be added in a follow up PR I think

Reply:
Also, the `Adaptive` name space does not have a `ComboBox` at this time